src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftAPIClient.java

```diff
@@ -2,194 +2,134 @@
-        ProcessBuilder builder = new ProcessBuilder("swift", "--os-auth-token", token, "--os-storage-url",
-                swiftUrl, "list", "-p", prefix, containerName);
-        LOGGER.debug("Executing command " + builder.command());
-
-        Process p;
-        String output;
-
-        try {
-            p = builder.start();
-            p.waitFor();
-
-            output = ProcessUtil.getOutput(p);
-
-            return getOutputLinesIntoList(output);
-        } catch (IOException e) {
-            LOGGER.error("Error while listing files from " + containerName);
-            return new ArrayList<String>();
-        } catch (InterruptedException e) {
-            LOGGER.error("Error while listing files from " + containerName);
-            return new ArrayList<String>();
+    public List<String> listFiles(String containerName, String dirPath) throws IOException {
+        List<String> files = new ArrayList<>();
+        String url = String.format(CONTAINER_URL_PATTERN, swiftUrl, containerName, dirPath);
+        HttpClient client = HttpClients.createDefault();
+        HttpGet httpget = new HttpGet(url);
+        httpget.addHeader("X-Auth-Token", token.getAccessId());
+        HttpResponse response = client.execute(httpget);
+        if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK && response.getStatusLine().getStatusCode() != HttpStatus.SC_NO_CONTENT) {
+            throw new IOException("The request to list files on object storage was failed: " + EntityUtils.toString(response.getEntity()));
         }
-    }
-
-    private List<String> getOutputLinesIntoList(String fileNames) throws IOException {
-        List<String> fileNamesList = new ArrayList<String>();
-
-        String[] lines = fileNames.split(System.getProperty("line.separator"));
-
-        for (int i = 0; i < lines.length; i++) {
-            fileNamesList.add(lines[i]);
+        if(Objects.nonNull(response.getEntity())) {
+            files = Arrays.asList(EntityUtils.toString(response.getEntity()).split("\n"));
         }
-
-        return fileNamesList;
+        return files;
     }
 
-    protected String generateToken() {
-
-        try {
-            ProcessBuilder builder = new ProcessBuilder("bash",
-                    properties.get(SapsPropertiesConstants.FOGBOW_CLI_PATH) + File.separator + "bin/fogbow-cli",
-                    "token", "--create", "-DprojectId=" + projectId, "-DuserId=" + userId, "-Dpassword=" + userPassword,
-                    "-DauthUrl=" + tokenAuthUrl, "--type", "openstack");
-
-            LOGGER.debug("Executing command " + builder.command());
-
-            Process p = builder.start();
-            p.waitFor();
-
-            return ProcessUtil.getOutput(p);
-        } catch (Throwable e) {
-            LOGGER.error("Error while generating keystone token", e);
+    public boolean existsTask(String containerName, String basePath, String taskId) throws IOException {
```

Comment:
isn't there a better way in the swift API to check if a file exists? It seems you are implementing a exists based on a list of a directory (I mean, maybe the API has a more direct option). Please, check the API. If there is a better way, create an issue to implement it latter.

Reply:
I checked and there is a more direct way to perform this check. I created the issue #100.

Reply:
ack